File: src/main/java/com/hileco/cortex/instructions/conditions/GREATER_THAN.java
package com.hileco.cortex.instructions.conditions;

import lombok.EqualsAndHashCode;

import java.math.BigInteger;
import java.util.Arrays;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
public class GREATER_THAN extends ConditionInstruction {
    @Override
    public boolean innerExecute(byte[] left, byte[] right) {
        var leftAsBigInteger = new BigInteger(left);
        var rightAsBigInteger = new BigInteger(right);
        return leftAsBigInteger.compareTo(rightAsBigInteger) > 0;
    }
}

File: src/main/java/com/hileco/cortex/instructions/conditions/GREATER_THAN.kt
package com.hileco.cortex.instructions.conditions

import com.hileco.cortex.constraints.expressions.Expression
import java.math.BigInteger

class GREATER_THAN : ConditionInstruction() {
    override fun innerExecute(left: ByteArray, right: ByteArray): Boolean {
        return BigInteger(left) > BigInteger(right)
    }

    override fun innerExecute(left: Expression, right: Expression): Expression {
        return Expression.GreaterThan(left, right)
    }
}
